### PR TITLE
Added import flow

### LINKED ISSUE #83: Develop /osio/import
Parent #47

### PR DESCRIPTION
**Pull request by edewit**
fixes #83

### CODE REVIEW

File: addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/OsioMissionControl.java [hunk 1]
```diff
@@ -65,39 +63,60 @@ public Boom launch(Projectile genericProjectile) throws IllegalArgumentException
...
-                    @Override
-                    public List<OpenShiftResource> getResources() {
-                        return null;
-                    }
-                })
+                .createdProject(new OsioOpenShiftProject(projectile))
                 .build();
     }
 
+    private static class OsioOpenShiftProject implements OpenShiftProject {
```
Comment (gastaldi) [inline addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/OsioMissionControl.java]: How about adding @Immutable to OpenShiftProject and use the generated ImmutableOpenShiftProject class instead?

File: addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/projectiles/OsioProjectile.java [hunk 2]
```diff
@@ -8,6 +8,9 @@
  */
 @Value.Immutable
 public interface OsioProjectile extends CreateProjectile {
+
+    String getGitOrganization();
```
Comment (gastaldi) [inline addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/projectiles/OsioProjectile.java]: That's ok, perhaps we should consider moving that to `CreateProjectile`

Comment (edewit) [reply addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/projectiles/OsioProjectile.java]: atm we don't support changing the organization for launch, but we could

**Comment by gastaldi** [comment]
@edewit let's make sure that https://github.com/openshiftio/openshift.io/issues/2358 is also fixed in this PR

File: addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/OsioMissionControl.java [hunk 3]
```diff
@@ -65,39 +59,37 @@ public Boom launch(Projectile genericProjectile) throws IllegalArgumentException
...
         GitRepository repository = gitSteps.createRepository(projectile);
 
+        executeCommonSteps(projectile, repository);
+
+        gitSteps.pushToGitRepository(projectile, repository);
+
+        openShiftSteps.triggerBuild(projectile);
+        return ImmutableBoom.builder()
+                .createdRepository(repository)
+                .createdProject(ImmutableOpenShiftProject.builder().name(projectile.getOpenShiftProjectName()).build())
```
Comment (gastaldi) [inline addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/OsioMissionControl.java]: Have you tested if this code works? The builder as it is defined right now requires you to specify a the consoleOverviewUrl. Unless you add a @Nullable to it

File: addons/new-osio-addon/src/test/java/io/fabric8/launcher/osio/steps/OpenShiftStepsIT.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                .addAsLibraries(Maven.resolver()
+                                        .loadPomFromFile("pom.xml")
+                                        .importDependencies(RUNTIME, COMPILE, TEST)
+                                        .resolve().withTransitivity().asFile());
+    }
+
+
+    @Test
+    public void shouldCreateBuildConfig() {
+        BuildConfig buildConfig = given().spec(configureEndpoint()).get().then().extract().body().as(BuildConfig.class);
```
Comment (gastaldi) [inline addons/new-osio-addon/src/test/java/io/fabric8/launcher/osio/steps/OpenShiftStepsIT.java]: Shouldn't this test do some sort of assertion?

**Comment by gastaldi** [comment]
Mind rebasing this PR?

**Comment by gastaldi** [comment]
[test]

File: addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/steps/GitSteps.java [hunk 5]
```diff
@@ -87,4 +90,11 @@ public void createWebHooks(OsioProjectile projectile, GitRepository gitRepositor
         }
         statusEvent.fire(new StatusMessageEvent(projectile.getId(), GITHUB_WEBHOOK));
     }
+
+    public GitRepository findRepository(OsioProjectile projectile) {
+        String repositoryName = projectile.getGitRepositoryName();
+        ImmutableGitOrganization gitOrganization = ImmutableGitOrganization.of(projectile.getGitOrganization());
+
+        return gitService.getRepository(gitOrganization, repositoryName).orElseThrow(IllegalArgumentException::new);
```
Comment (gastaldi) [inline addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/steps/GitSteps.java]: Better throw a meaningful message here

**Pull request merged**